core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -261,6 +262,16 @@ public Snapshot snapshot(long snapshotId) {
     return lazyTable().history();
   }
 
+  @Override
+  public Map<String, SnapshotRef> refs() {
+    return lazyTable().refs();
```

Comment:
Seems reasonable to read the table metadata for this. I doubt serialized tables will be accessing refs.